### ISSUE TITLE
File upload on modular forms with hash

### ISSUE THREAD

**Issue by david-bettondesign** [getgrav/grav-plugin-form#192]
This issue is quite specific but it's an issue none the less. This issue affects front end forms on modular pages.

**Steps to reproduce:**

1. Create a modular page
2. Add a form to one of the pages
3. Add a `file` field to the form
4. In the form action put something like `'/my-page#contact'` (the page plus an anchor).
5. Submit the form as a user

**Expected result:**
The form submits all of the data and uploads the file to the directory you have specified. If using Email then the file should also be attached. 

**Actual result:**
The form submits all of the data **except for** the file upload. Nothing uploads and if using Email then all of the data is sent **except for** the file upload.

I know this is a little obscure but I was working on it for many hours until I figured out that the file wasn't uploading because of the anchor.

**Comment by tutchmedia** [comment]
Wow, just found this after 2 hours - same exact problem! In my case i was passing a career to the form...

Sure enough removing it made the attachment work. Did you get a workaround at all?

**Comment by rhukster** [comment]
I've not dug into this one yet, but the weird thing is we don't actually get anchor tags sent to the server, they are a client-side only thing.

**Comment by tutchmedia** [comment]
It is strange - it's literally stopping us deploying to live :(

In my case the URL looks likes /job-application/career:my_first_career. We then bring that career title into the form for submitting alongside the application and name.

**Comment by tutchmedia** [comment]
Ok - so upon further inspection, it appears that when the URL has anything other than the default page uri, the dropzone uploader will try to fetch an invalid .json file on upload complete.

In my example above, instead of fetching the default career-apply.json which will work, it's instead looking for career:level-designer.json which returns the oops, something went wrong html.

The culprit line is in form.min.js:204 it seems:
`url: e.data("file-url-add") || (e.closest("form").attr("action") || d.config.current_url) + ".json",`

Had to change it to use only the current_url for now but thought i'd help towards a permanent fix.

**Comment by tutchmedia** [comment]
@rhukster - Sorry to be a pain but is this something we could get sorted soon? Currently having to manually fix every update so would be amazing if a permanent fix was applied.

**Comment by sebastianbaumann** [comment]
Thanks for the bug report. Having the same issue right now. 
Any news on fixing this?

**Comment by tutchmedia** [comment]
Just took a few updates that have been hanging around for a while and fell into this trap (as they removed my workaround). Would be great to get this resolved @rhukster please? :)

**Comment by mahagr** [comment]
@tutchmedia Did you have some fix for this one?

**Comment by tutchmedia** [comment]
@mahagr - The issue is basically that when we have the following '/job-application/career:my_first_career', the file upload will try to upload to career:job-application.json resulting in a 404 error. We need to instead go to job-application.json.

**Comment by mahagr** [comment]
Yes, the same is true for `?foo=bar` as well, there's no logic to check that there are no grav/query params used in the URL.

**Comment by mahagr** [comment]
@tutchmedia Just in case, can you test if we haven't fixed this already in Grav 1.6 (and Form 3.0) RCs?

**Comment by tutchmedia** [comment]
@mahagr - sadly i won't be able to check by installing it but having a look at the Form 3.0 RC code, it looks to have the same logic on line 177 of assets/form.min.js 

`  url: container.data('file-url-add') || (container.closest('form').attr('action') || _gravForm.config.current_url) + '.json',`

Hope this helps
